src/ar_gateway_server.erl

```diff
@@ -0,0 +1,36 @@
+-module(ar_gateway_server).
+-export([start/3]).
+
+%%%===================================================================
+%%% Public interface.
+%%%===================================================================
+
+%% @doc Start the server that will handle gateway requests.
+start(Port, Domain, CustomDomains) ->
+	ProtocolOpts = #{
+		middlewares => [cowboy_handler],
```

Comment:
Shoudn't we have the `ar_firewall` here?

Reply:
So as agreed not for now